### PR TITLE
Consolidated-improvements

### PR DESCRIPTION
**Pull request by ghislaineguerin**
### Modals Update

I've been improving modals across Mathesar to ensure consistency, better usability, and adherence to design guidelines. Here's a summary of the changes and improvements:

#### Alignment and Spacing Issues
- **Modal Headers:** The prominence of modal headers has been increased to ensure balance with the footers, as there was insufficient visual weight at the top.
- image
- The modal title size is now consistent and larger than content headings. This issue was especially noticeable in the Constraints modal, where the headers inside the modal were larger than the modal header.
image
- **Close Button Alignment:** The close button is now correctly aligned with the start of the title in all modals.
- image
- **Collapsing Headers:** In Constraint modals, collapsing headers were adjusted to maintain a consistent height, regardless of whether a button is present.
- image

#### Styling and Contrast Issues
- **Form Field Labels:** Improved the contrast of form field labels to distinguish them clearly from body text.
- image
- **Message Boxes:** Updated the style of message boxes to have a less boxy appearance, improving the flow of content and reducing vertical separations that make content harder to scan.
- image
- **Inline Forms and Descriptive Button Labels:** Inline forms were used to add foreign or unique keys, and redundant titles were removed. Instead, the save button labels were updated to indicate the action, such as adding a new object.
- image

#### Consistency Enhancements
- **Input Box Style:** Ensured consistency in using boxed input groups, applying the same style across different modals, such as import modals and database creation modals.
    - image
- **Modal Descriptions:** Standardized the placement and presentation of descriptions.xw

### Create Link Updates

The "Create Link" modal has been revised to enhance clarity and maintain consistency with other forms in the app.

- **Outcome Box Update**: Each outcome now has its own collapsible section containing a single, concise outcome statement. Previously, the outcome box listed all the input boxes for naming every element involved, which cluttered the interface. If the user agrees with the default outcomes, they won't need to interact with these input boxes. This change reduces visual distractions and makes it easier to review the outcomes which was the main goal for this special message box.
- **Updated Labels**: Labels have been revised to follow the new naming convention for new elements. For example, fields now use phrasing like "Name of new column" instead of "New column name".
- image
- image

### Color Scheme

- The color scheme has been updated to increase contrast. This change applies to the inspector panel, message boxes, and buttons.
image

### Record Summary Section

- The record summary section was updated for consistent language and to reduce redundant elements.
- Spacing between elements has been fixed, as some were too close together, such as the "Append Column" button and the input box.
- The tint for the record summary element, originally specified as dark yellow, lacked sufficient yellow and was adjusted from an orange tone to a more vibrant color.
- image

### Form Style Updates

- Labels have been made slightly bolder. Spacing between labels and input boxes has been updated for consistency with other parts of the app.
- image

### Button Sizes

- Button sizes have been updated so that all are using the "small" style when inside inspector.

### Spacing for empty states in tabs

- Spacing for empty states has been updated to be consistent across all tabs.
- image

### Links Styling
- Link card styling updated to match record link color shade. Direction icons moved outside circles to reduce visual weight.
- image

### General Improvements

- General spacing fixes for consistency
- image
- Apply consistent use of bold styles
- image
- Removed special layout width that was only used for record page. There's not a particular issue that is solved by making the container narrower. It will be more consistent to have the same width as other containers.
- image

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208365459259464

### CODE REVIEW

**Review by seancolsen — CHANGES_REQUESTED** [review]
Looking great!

I added a couple comments on the diff.

And I have some other general comments...

---

### Global CSS

In general I try to avoid global CSS whenever it's relatively easy to do so. Global CSS adds tight coupling between components that's difficult to spot when making changes later.

In `TableInspector.svelte` you've added some global CSS that doesn't need to be global. It would be good to switch that to scoped CSS.

You've added three new global CSS selectors in `LinkTableForm.svelte` that I don't really like. I think you ought to be able to get rid of the global CSS by doing the following instead:

1. To adjust the font weight on the collapsible headers, use this CSS in LinkTableForm.svelte

    ```scss
    .form {
      // ...
      --Collapsible_header-font-weight: var(--font-weight-medium);
    }
    ```

    This works because the Collapsible component is already set up with that CSS variable exactly where we need it here.

    Less code too!

1. To add some space around the content in each collapsible section, wrap the content in a div.

    ```html
    <Collapsible>
      <!-- ... -->
      <svelte:fragment slot="content">
        <div class="collapsible-detail"> <!-- 👈 ADDED ELEMENT  -->
          <Field field={mappingTableName} label={$_('table_name')} />
        <div> 
      </svelte:fragment>
    </Collapsible>
    ```

    (You'd need to add a div like that for each of the three collapsible sections you have.)

    Then at the bottom, use this CSS to refer to that div in the same component.

    ```scss
    .collapsible-detail {
      padding: var(--size-xx-small);
      margin-top: 0.25em;
    }
    ```

    Yes, this is slightly more code, but it keeps the CSS scoped which is less bug prone. In theory you could create a whole new component to handle this sort of code reuse, but that's probably overkill. In Svelte 5, I think snippets will help to solve this problem somewhat.

1. To add some space between successive collapsible sections, add a wrapper element like this:

    ```html
    <OutcomeBox>
      <div class="collapsible-sections"> <!-- 👈 ADDED ELEMENT  -->
        <!-- ≈100 lines of markup -->
      </div>
    </OutcomeBox>
    ```

    And CSS like this:

    ```scss
    .collapsible-sections > :global(* + *) {
      margin-top: var(--size-xx-small);
    }
    ```

    This one is more borderline though. Not as important. If you don't want to make this change, that's fine with me. In your current code, the only thing you're coupling in Collapsible is the top-level CSS name, which is very unlikely to change. But I mention the above suggestion in hopes that it will give you some ideas for how to think in terms of scoped CSS for other cases like this in the future. Yes, the CSS in my suggestion above still has the `:global` syntax, but it's not referencing any selectors defined in other components, so I consider it to be more resilient to future refactoring.

⚠️ **If** you want to _keep_ some of the global CSS, it is somewhat important to change the selectors a little...

Instead of something like `:global(.form .collapsible-content)`, it's much better to use `.form :global(.collapsible-content)`. The first syntax sets that CSS for anything anywhere in the whole app. This could cause bugs if other code in other parts of the app happen to use that same class structure. The second syntax uses global CSS but still keeps it contained within this component. It can still lead to bugs if the Collapsible component is later refactored. But the risk of bugs cropping up is significantly lower.



### Link colors

In the CreateLink dialog, the colors of these table pills need to be swapped to match the colors of the pills in the diagram.

image

**Comment by seancolsen** [comment]
Also @ghislaineguerin a couple requests to smooth the PR review process:

- When you're ready for review, please remember to assign me to the PR and label the PR with `pr-status: review`. I know that GitHub has the "request review" feature, but the workflow that I'm accustomed on our team is to use assignments and labels instead.

- Also, please don't click "Resolve conversation" on other people's review comments. It _is_ helpful if you can comment on each one linking to the commit where you made the fix. That saves me time when I re-review because it's easy to verify that you've addressed my critique. You can paste the hash of the commit into the GitHub comment and GitHub will turn it into a link.

File: mathesar_ui/src/component-library/window/Window.scss [hunk 1]
```diff
@@ -3,6 +3,7 @@
   border-radius: 0.4em;
   min-height: var(--window-min-height, 7em);
   width: var(--window-width, 100%);
+  min-width: var(--window-min-width, 40em);
```
[the remark this answers is not in the document]
Comment (ghislaineguerin) [reply mathesar_ui/src/component-library/window/Window.scss:6]: @seancolsen I wanted the record selector to have a min-width, specially when it's a one column table. We could set it to that component instead. But it is not critical and I can just revert that change.

File: mathesar_ui/src/components/message-boxes/OutcomeBox.svelte [hunk 2]
```diff
@@ -8,16 +8,19 @@
   type $$Props = ComponentProps<MessageBox>;
 
   export let fullWidth = true;
+  export let title = "What's Changing";
```
[the remark this answers is not in the document]
Comment (ghislaineguerin) [reply mathesar_ui/src/components/message-boxes/OutcomeBox.svelte]: https://github.com/centerofci/mathesar/commit/f180944f5cf763a45091ea01033d02366f0e30cd

File: mathesar_ui/src/component-library/window/Window.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (ghislaineguerin) [reply mathesar_ui/src/component-library/window/Window.scss]: https://github.com/centerofci/mathesar/commit/80e4cc51ce6caef3238f65b7713a993531868af6

**Comment by ghislaineguerin** [comment]
> In the CreateLink dialog, the colors of these table pills need to be swapped to match the colors of the pills in the diagram.

@seancolsen this is also present on current develop. Do you have a fix I could apply? I'm thinking that the new column component needs to be aware of which is the current color for base and apply the opposite.

**Review by seancolsen — APPROVED** [review]
Looks good!

**Comment by seancolsen** [comment]
@ghislaineguerin thanks for identifying that the table pill color bug was not actually a regression. I opened #3886 to track it. I don't think we need to worry about it for now.

**Pull request merged**